src/rgw/rgw_common.h

```diff
@@ -1435,6 +1436,11 @@ class RGWEnv;
 /* Namespaced forward declarations. */
 namespace rgw {
   namespace auth {
+    class Identity;
+    // size of static storage needed to fit any derived Identity
+    static constexpr size_t MaxIdentitySize = 704;
```

Comment:
@cbodley how did we reach that number, could that number change? If so, maybe we need to dynamically create it in compilation if possible or do some kind of a test / assert that we have the right number here?

Reply:
@yehudasa: conceptually there should be a `static_assert` in static pointer implementation disallowing assignment to an instance with too small internal storage.

I expect the maximum needed capacity depends on architecture and, theoretically, even on compiler.

Reply:
> I expect the maximum needed capacity depends on architecture and, theoretically, even on compiler.

That's why we should try to find a way to generate this value during compilation